easybuild/framework/easyblock.py

```diff
@@ -3350,6 +3352,271 @@ def _sanity_check_step_multi_deps(self, *args, **kwargs):
+                summary_msg += f"{num_files_surplus_cc_fails})\n"
+            else:
+                summary_msg += f"{num_files_surplus_cc} (not running with --strict-cuda-sanity-check, so not "
+                summary_msg += "considered failures)\n"
+            summary_msg += "Number of files missing PTX code for the highest configured CUDA Compute Capability: "
+            if accept_missing_ptx:
+                summary_msg += f"{num_files_missing_ptx} (running with --accept-missing-cuda-ptx so not considered "
+                summary_msg += "failures)\n"
+            else:
+                summary_msg += f"{num_files_missing_ptx} (ignored: {num_files_missing_ptx_ignored}, fails: "
+                summary_msg += f"{num_files_missing_ptx_fails})\n"
+            if not build_option('debug'):
+                summary_msg += "Rerun with --debug to see a detailed list of files.\n"
+            # Give some advice
+            if num_files_missing_cc > 0 and not accept_ptx_as_cc:
+                summary_msg += "\nYou may consider rerunning with --accept-ptx-as-cc-support to accept binaries that "
+                summary_msg += "don't contain the device code for your requested CUDA Compute Capabilities, but that "
+                summary_msg += "do have PTX code that can be compiled for your requested CUDA Compute "
+                summary_msg += "Capabilities. Note that this may increase startup delay due to JIT compilation "
+                summary_msg += "and may also lead to suboptimal runtime performance, as the PTX code may not exploit "
+                summary_msg += "all features specific to your hardware architecture.\n"
+            if num_files_surplus_cc > 0 and strict_cc_check:
+                summary_msg += "\nYou may consider running with --disable-strict-cuda-sanity-check. This means you'll "
+                summary_msg += "accept that some binaries may have CUDA Device Code for more architectures than you "
+                summary_msg += "requested, i.e. the binary is 'fatter' than you need. Bigger binaries may generally "
+                summary_msg += "cause some startup delay, and code path selection could introduce a small overhead, "
+                summary_msg += "though this is generally negligible.\n"
+            if num_files_missing_ptx > 0 and not accept_missing_ptx:
+                summary_msg += "\nYou may consider running with --accept-missing-cuda-ptx to accept binaries that "
+                summary_msg += "don't contain PTX code for the highest CUDA Compute Capability you requested. This "
+                summary_msg += "breaks forwards compatibility for newer CUDA Compute Capabilities (i.e. your compiled "
+                summary_msg += "binaries will not run on cards with higher CUDA Compute Capabilities than what "
+                summary_msg += "you requested in --cuda-compute-capabilities), but that may be acceptable to you.\n"
+            self.log.info(summary_msg)
```

Comment:
Perhaps if `ignore_failures` is set we actually print the `summary_msg` as a warning?

Reply:
I think that's actually a great idea (though note that we already print warning for the individual things that would have been a failure).